electrum/gui/qt/network_dialog.py

```diff
@@ -227,344 +214,109 @@ def update(self):
-            if not server: raise Exception("failed to parse")
-        except Exception:
-            return
-        net_params = net_params._replace(server=server,
-                                         auto_connect=self.autoconnect_cb.isChecked())
-        self.network.run_from_another_thread(self.network.set_parameters(net_params))
-
-    def set_proxy(self):
         net_params = self.network.get_parameters()
         if self.proxy_cb.isChecked():
             if not self.proxy_port.hasAcceptableInput():
                 return
-            proxy = {'mode':str(self.proxy_mode.currentText()).lower(),
-                     'host':str(self.proxy_host.text()),
-                     'port':str(self.proxy_port.text()),
-                     'user':str(self.proxy_user.text()),
-                     'password':str(self.proxy_password.text())}
+            proxy = {'mode': str(self.proxy_mode.currentData()),
+                     'host': str(self.proxy_host.text()),
+                     'port': str(self.proxy_port.text()),
+                     'user': str(self.proxy_user.text()),
+                     'password': str(self.proxy_password.text())}
         else:
             proxy = None
-            self.tor_cb.setChecked(False)
         net_params = net_params._replace(proxy=proxy)
         self.network.run_from_another_thread(self.network.set_parameters(net_params))
 
-    def _on_tab_changed(self):
-        if self.tabs.currentWidget() is self._proxy_tab:
-            self.td.trigger_rescan()
-
-    def suggest_proxy(self, found_proxy):
-        if found_proxy is None:
-            self.tor_cb.hide()
-            return
-        self.tor_proxy = found_proxy
-        self.tor_cb.setText(_("Use Tor proxy at port {}").format(str(found_proxy[1])))
-        if (self.proxy_cb.isChecked()
-                and self.proxy_mode.currentIndex() == self.proxy_mode.findText('SOCKS5')
-                and self.proxy_host.text() == "127.0.0.1"
-                and self.proxy_port.text() == str(found_proxy[1])):
-            self.tor_cb.setChecked(True)
-        self.tor_cb.show()
-
-    def use_tor_proxy(self, use_it):
-        if not use_it:
-            self.proxy_cb.setChecked(False)
-        else:
-            socks5_mode_index = self.proxy_mode.findText('SOCKS5')
-            if socks5_mode_index == -1:
-                _logger.info("can't find proxy_mode 'SOCKS5'")
-                return
-            self.proxy_mode.setCurrentIndex(socks5_mode_index)
-            self.proxy_host.setText("127.0.0.1")
-            self.proxy_port.setText(str(self.tor_proxy[1]))
-            self.proxy_user.setText("")
-            self.proxy_password.setText("")
-            self.tor_cb.setChecked(True)
-            self.proxy_cb.setChecked(True)
-        self.check_disable_proxy(use_it)
-        self.set_proxy()
-
-    def proxy_settings_changed(self):
-        self.tor_cb.setChecked(False)
-
-
-class TorDetector(QThread):
```

Comment:
> The separate Tor checkbox was removed, as it was somewhat confusing and not kept in UI-sync with actual Tor use.

The utility here was that if the user had a Tor proxy available they could use, e.g. because their Linux distro runs one in the background, or because they have Tor Browser running, they could set that up using a single click. I think this was useful, but this PR removes it completely AFAICT :/

It does not have to be a checkbox, it could be a button too: "Tor proxy found: [activate]", or maybe in a completely different way.

Reply:
Maybe something like this: when user enables the 'use proxy' checkbox, it does a probe. If it finds a proxy on one of the ports (also non-tor proxies, but e.g. prefer tor), show a yes/no popup asking the user to apply the detected settings. The probe should maybe only happen if the user did not already enter a hostname and/or port number, so this is not spammy. 

Note: I think we should keep proxy settings stored in config, even if proxy gets disabled. Currently all proxy related settings are deleted from config when proxy gets disabled.